Sources/BSON/BSONDocument.swift

```diff
@@ -150,11 +153,9 @@ public struct BSONDocument {
             return nil
         }
         set {
-            do {
-                try self.set(key: key, of: newValue)
-            } catch {
-                fatalError("\(error)")
-            }
+            // The only time this would crash is document too big error
```

Comment:
This would change from the fatalError behavior in the driver. It's not exactly something one could rely on (like API wise) but going from crash to silence doesn't seem right here, thoughts?

Reply:
Oh if that is the current behavior of the driver, then we should preserve it for now. I filed [SWIFT-892](https://jira.mongodb.org/browse/SWIFT-892) so we can talk as a team about better approaches to this.